library/L2_HAL/displays/lcd/st7066u.hpp

```diff
@@ -167,56 +143,46 @@ class St7066u
   ///
   /// @param operation Operation transfer type.
   /// @param nibble    4-bit data to transfer.
-  void WriteNibble(WriteOperation operation, uint8_t nibble)
+  void WriteNibble(WriteOperation operation, uint8_t nibble) const
   {
-    kControlPins.e.Set(sjsu::Gpio::State::kHigh);
-    kControlPins.rs.Set(sjsu::Gpio::State(operation));
-    kControlPins.rw.Set(sjsu::Gpio::State::kLow);
-    // set nibble on 4-bit data bus
-    kControlPins.d7.Set(sjsu::Gpio::State((nibble >> 3) & 0x01));
-    kControlPins.d6.Set(sjsu::Gpio::State((nibble >> 2) & 0x01));
-    kControlPins.d5.Set(sjsu::Gpio::State((nibble >> 1) & 0x01));
-    kControlPins.d4.Set(sjsu::Gpio::State((nibble >> 0) & 0x01));
+    kEnablePin.Set(sjsu::Gpio::State::kHigh);
+    kRegisterSelectPin.Set(sjsu::Gpio::State(operation));
+    kReadWritePin.Set(sjsu::Gpio::State::kLow);
+
+    kDataBus.Write(nibble);
     sjsu::Delay(1ms);
     // Toggle chip enable to trigger write on falling edge
-    kControlPins.e.Set(sjsu::Gpio::State::kLow);
-    kControlPins.e.Set(sjsu::Gpio::State::kHigh);
+    kEnablePin.Set(sjsu::Gpio::State::kLow);
+    kEnablePin.Set(sjsu::Gpio::State::kHigh);
   }
 
   /// Transfers a command or data byte to the device.
   ///
   /// @param operation Operation transfer type.
   /// @param byte      Byte to transfer.
-  void WriteByte(WriteOperation operation, uint8_t byte)
+  void WriteByte(WriteOperation operation, uint8_t byte) const
```

Comment:
Is WriteByte or WriteNibble actually any different? They look basically the same now that we are using ParallelBus. You might be able to consolidate them together and simply check if you need to run it twice for a nibble in the places where WriteByte/Nibble is used. 

If this works, then you can reduce the code size of the object.